src/System.IO.Packaging/tests/Tests.cs

```diff
@@ -3639,6 +3639,109 @@ public void SetEmptyPropertyToNull()
             }
         }
 
+        [Fact]
+        public void CreateWithFileAccessWrite()
+        {
+            string[] fileNames = new [] { "file1.txt", "file2.txt", "file3.txt" };
+
+            using (var stream = new MemoryStream())
+            {
+                using (var package = Package.Open(stream, FileMode.Create, FileAccess.Write))
+                {
+                    foreach( var fileName in fileNames)
+                    {
+                        PackagePart part = package.CreatePart(PackUriHelper.CreatePartUri(new Uri(fileName, UriKind.Relative)),
+                                                              System.Net.Mime.MediaTypeNames.Text.Plain,
+                                                              CompressionOption.Fast);
+                        using (var writer = new StreamWriter(part.GetStream(), Encoding.ASCII))
+                        {
+                            // just write the filename as content
+                            writer.Write(fileName);
+                        }
+                    }
+                }
+
+                // reopen for read and validate the content
+                stream.Seek(0, SeekOrigin.Begin);
+                using (var readPackage = Package.Open(stream))
+                {
+                    foreach(var fileName in fileNames)
```

Comment:
Nit: var => string... will stop commenting on these :)

Reply:
yeah, sorry about that, I had cleaned it up but then reverted to an old version.